dev-client/src/screens/ProjectListScreen/components/ProjectList.tsx

```diff
@@ -16,20 +16,49 @@
  */
 
 import {Box, FlatList, Text} from 'native-base';
+import {useCallback} from 'react';
 import {useTranslation} from 'react-i18next';
 
 import {Project} from 'terraso-client-shared/project/projectSlice';
 import {useListFilter} from 'terraso-mobile-client/components/ListFilter';
 import {ProjectPreviewCard} from 'terraso-mobile-client/screens/ProjectListScreen/components/ProjectPreviewCard';
 
+// TODO(performance):
+// Some relevant reading: https://reactnative.dev/docs/optimizing-flatlist-configuration#windowsize
+const WINDOW_SIZE = 6; // Default is 21, bringing it down as the items in this FlatList are very costly to render
+const MAX_TO_RENDER_PER_BATCH = 5; // Similar as above, default is 10
+// const ITEM_HEIGHT = ??
+const SEPARATOR_HEIGHT = 8;
+
+type RenderItemProps = {
+  item: Project;
+};
+
 export const ProjectList = () => {
   const {t} = useTranslation();
   const {filteredItems} = useListFilter<Project>();
+
+  const renderItem = useCallback(
+    ({item}: RenderItemProps) => <ProjectPreviewCard project={item} />,
+    [],
+  );
+
+  // TODO(performance): Consider adjusting ProjectPreviewCard with the design team
+  // to always have a fixed height so you can leverage getItemLayout (example below):
+  // const getItemLayout = useCallback((_: any, index: number) => {
+  //   const length = ITEM_HEIGHT + SEPARATOR_HEIGHT;
+  //   const offset = length * index;
+  //   return {length, offset, index};
+  // }, []);
```

Comment:
this is kind of a general UI question, but if you're specifying fixed heights in pixels in advance, how do you handle something like a user who has set a large font size preference for accessibility reasons?

Reply:
Need to look into that, but off the top of my head, I think it doesn't have to be fixed-fixed but rather deterministic and repeated for all items, great question though

Reply:
i see, so it'd work but you'd need to find a way to compute that dynamically. i've run into this problem with other UI frameworks before, never understood why there can't be an option to say "i promise they'll all be the _same_ dimension" instead of "i promise they'll all be _this_ dimension" so the framework can just measure one item's layout and use that value for the rest of the list which seems much less error-prone